File: src/backend/cadastros/src/model/pais_model.py
from src import db


class PaisModel(db.Model):
    __tablename__ = 'pais'

    id = db.Column(db.Integer, primary_key=True)
    nome_ptbr = db.Column(db.String(100))
    nome_en = db.Column(db.String(100))
    codigo = db.Column(db.Integer)
    sigla2 = db.Column(db.String(2))
    sigla3 = db.Column(db.String(3))
    codigo_bacen = db.Column(db.Integer)


    def mapping(self, data):
        self.id = data.get('id')
        self.nome_ptbr = data.get('nomePtbr')
        self.nome_en = data.get('nomeEn')
        self.codigo = data.get('codigo')
        self.sigla2 = data.get('sigla2')
        self.sigla3 = data.get('sigla3')
        self.codigo_bacen = data.get('codigoBacen')

    def serialize(self):
        return {
            'id': self.id,
            'nomePtbr': self.nome_ptbr,
            'nomeEn': self.nome_en,
            'codigo': self.codigo,
            'sigla2': self.sigla2,
            'sigla3': self.sigla3,
            'codigoBacen': self.codigo_bacen,
        }

File: frontend/cadastros/lib/app/data/model/pais_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';


class PaisModel {
	int? id;
	String? nomePtbr;
	String? nomeEn;
	int? codigo;
	String? sigla2;
	String? sigla3;
	int? codigoBacen;

	PaisModel({
		this.id,
		this.nomePtbr,
		this.nomeEn,
		this.codigo,
		this.sigla2,
		this.sigla3,
		this.codigoBacen,
	});

	static List<String> dbColumns = <String>[
		'id',
		'nome_ptbr',
		'nome_en',
		'codigo',
		'sigla2',
		'sigla3',
		'codigo_bacen',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Nome Ptbr',
		'Nome En',
		'Codigo',
		'Sigla2',
		'Sigla3',
		'Codigo Bacen',
	];

	PaisModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		nomePtbr = jsonData['nomePtbr'];
		nomeEn = jsonData['nomeEn'];
		codigo = jsonData['codigo'];
		sigla2 = jsonData['sigla2'];
		sigla3 = jsonData['sigla3'];
		codigoBacen = jsonData['codigoBacen'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['nomePtbr'] = nomePtbr;
		jsonData['nomeEn'] = nomeEn;
		jsonData['codigo'] = codigo;
		jsonData['sigla2'] = sigla2;
		jsonData['sigla3'] = sigla3;
		jsonData['codigoBacen'] = codigoBacen;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		nomePtbr = plutoRow.cells['nomePtbr']?.value;
		nomeEn = plutoRow.cells['nomeEn']?.value;
		codigo = plutoRow.cells['codigo']?.value;
		sigla2 = plutoRow.cells['sigla2']?.value;
		sigla3 = plutoRow.cells['sigla3']?.value;
		codigoBacen = plutoRow.cells['codigoBacen']?.value;
	}	

	PaisModel clone() {
		return PaisModel(
			id: id,
			nomePtbr: nomePtbr,
			nomeEn: nomeEn,
			codigo: codigo,
			sigla2: sigla2,
			sigla3: sigla3,
			codigoBacen: codigoBacen,
		);			
	}

	
}